lib/puppet/settings.rb

```diff
@@ -1158,9 +1158,13 @@ def issue_deprecation_warning(setting, msg = nil)
       msg << " #{ref}" if ref
       Puppet.deprecation_warning(msg)
     when setting.completely_deprecated?
-      Puppet.deprecation_warning("Setting #{name} is deprecated. #{ref}", "setting-#{name}")
+      message = _("Setting %{name} is deprecated.") % { name: name }
+      message += " #{ref}"
+      Puppet.deprecation_warning(message, "setting-#{name}")
     when setting.allowed_on_commandline?
-      Puppet.deprecation_warning("Setting #{name} is deprecated in puppet.conf. #{ref}", "puppet-conf-setting-#{name}")
+      message = _("Setting %{name} is deprecated in puppet.conf.") % { name: name }
```

Comment:
`puppet.conf` is a filename and shouldn't be translated

Reply:
fixed